rclcpp/include/rclcpp/strategies/allocator_memory_strategy.hpp

```diff
@@ -507,6 +513,8 @@ class AllocatorMemoryStrategy : public memory_strategy::MemoryStrategy
   VectorRebind<std::shared_ptr<const rcl_timer_t>> timer_handles_;
   VectorRebind<std::shared_ptr<Waitable>> waitable_handles_;
 
+  std::map<std::shared_ptr<const rcl_service_t>, rclcpp::ServiceBase::WeakPtr> service_map_;
```

Comment:
do we really need this additional data-structure?
from what I see, there is now no need at all to store `rcl_service_t` in this file.
 - it's not needed anymore by the `add_to_wait_set` operation
 -  we use it only in a call to `get_service_by_handle`, which returns a `rclcpp::ServiceBase::SharedPtr` object.

I think that we can really treat services as waitables, where we have that `waitable_handles_` stores `std::shared_ptr<Waitable>`.

Reply:
In fact, I was also trying to unify it as a waitable , but I had to change the executor's service processing routine as well, so I wanted to check if this was okay. Does it mean to 1) remove execute_service() and get_next_service() from executor, and 2) process it within the service with execute() of waitable?

Reply:
Why do you think we have to change also those functions?
From what I see:
 - `execute_service` takes as input a `rclcpp::ServiceBase::SharedPtr`. Yes, we could definitely move this into the service itself, but I think it's out of the scope of this PR.
 - `get_next_service` would remain and it would just become similar to `get_next_waitable`

Reply:
Ah forget it. I didn't see the last sentence and I misunderstood that service should be integrated into waitable_handles_.

> _`I think that we can really treat services as waitables, where we have that waitable_handles_ stores std::shared_ptr<Waitable>.
> This means changing service_handles_ to store rclcpp::ServiceBase::SharedPtr_

Let me modify to store ServiceBase::SharedPtr instead of storing rcl handle in service_handles_.